ethereum/contracts/DOTApp.sol

```diff
@@ -46,21 +63,28 @@ abstract contract BaseDOTApp {
             _channelId == ChannelId.Incentivized,
             "Invalid channel ID"
         );
-        require(_amount % granularity() == 0, "Invalid Granularity");
+        require(_amount % granularity == 0, "Invalid Granularity");
 
         token.burn(msg.sender, _amount, abi.encodePacked(_recipient));
 
         OutboundChannel channel = OutboundChannel(channels[_channelId].outbound);
 
         bytes memory call = encodeCall(msg.sender, _recipient, unwrap(_amount));
-        channel.submit(call);
+        channel.submit(msg.sender, call);
     }
 
     function mint(bytes32 _sender, address _recipient, uint128 _amount) public {
         // TODO: Ensure message sender is a known inbound channel
         token.mint(_recipient, wrap(_amount), abi.encodePacked(_sender));
     }
 
+    function burnFee(address _account, uint256 _amount) external returns (uint128) {
```

Comment:
Note that this and other DOT-related changes is being updated/modified/removed in this follow-up PR: #309